src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.InstallRecords.cs

```diff
@@ -0,0 +1,339 @@
+                "Microsoft.NET.Sdk.macOS",
+                "Microsoft.NET.Sdk.Maui",
+                "Microsoft.NET.Sdk.tvOS",
+                "Microsoft.NET.Workload.Emscripten.Current",
+                "Microsoft.NET.Workload.Emscripten.net6",
+                "Microsoft.NET.Workload.Emscripten.net7",
+                "Microsoft.NET.Workload.Mono.ToolChain.Current",
+                "Microsoft.NET.Workload.Mono.ToolChain.net6",
+                "Microsoft.NET.Workload.Mono.ToolChain.net7",
+            })
+            {
+                manifestIDCasing[casedManifestID.ToLowerInvariant()] = casedManifestID;
+            }
+
+            string sdkManifestFolder = Path.Combine(DotNetHome, "sdk-manifests");
+
+            foreach (var manifestFeatureBandFolder in Directory.GetDirectories(sdkManifestFolder))
+            {
+                if (!ReleaseVersion.TryParse(Path.GetFileName(manifestFeatureBandFolder), out ReleaseVersion releaseVersion))
+                {
+                    //  Ignore folders which aren't valid version numbers
+                    Log.LogMessage($"Skipping invalid feature band version folder: {manifestFeatureBandFolder}");
+                    continue;
+                }
+                if (releaseVersion.Major < 8)
+                {
+                    //  Ignore manifests prior to 8.0.100, they were not side-by-side
+                    continue;
+                }
+                var manifestFeatureBand = new SdkFeatureBand(releaseVersion);
+
+                foreach (var manifestIDFolder in Directory.GetDirectories(manifestFeatureBandFolder))
+                {
+                    var lowerCasedManifestID = Path.GetFileName(manifestIDFolder);
+                    if (manifestIDCasing.TryGetValue(lowerCasedManifestID, out string manifestID))
+                    {
+                        foreach (var manifestVersionFolder in Directory.GetDirectories(manifestIDFolder))
+                        {
+                            string manifestVersionString = Path.GetFileName(manifestVersionFolder);
+
+                            if (discoveredManifests.Contains((manifestID, manifestVersionString)))
+                            {
+                                continue;
+                            }
+
+                            Log.LogMessage($"Discovered manifest installation in {manifestVersionFolder} which didn't have corresponding installation record in Registry");
+
+                            if (NuGetVersion.TryParse(manifestVersionString, out NuGetVersion manifestVersion))
+                            {
+                                var packageIdentity = new PackageIdentity(manifestID + ".Manifest-" + manifestFeatureBand, manifestVersion);
+                                string uuidName = $"{packageIdentity};{HostArchitecture}";
+                                var upgradeCode = '{' + CreateUuid(UpgradeCodeNamespaceUuid, uuidName).ToString() + '}';
```

Comment:
Can probably simplify this as `var upgradeCode = $"{{{CreateUuid(UpgradeCodeNamespaceUuid, uuidName)}}}";`

Reply:
To me that's not actually simpler, with all the repeated braces, two to escape a brace and another one for the string interpolation.